pkg/sql/opt/exec/execbuilder/testdata/lookup_join

```diff
@@ -815,7 +815,7 @@ query TTT
 EXPLAIN SELECT count(*) FROM (SELECT * FROM b NATURAL INNER LOOKUP JOIN a)
 ----
 ·                      distributed  true
-·                      vectorized   false
+·                      vectorized   true
```

Comment:
@yuzefovich This changed because the check I removed.. I am guessing that the code treats the error as "do not vectorize". I don't think it's right to catch all errors like this though, this was an assertion error so it should be visible.

Reply:
Yeah, we first attempt to "plan" a vectorized flow ("plan" is in quotes since we do a partial planning), and if any error is encountered, we fall back to row-by-row flow, regardless whether it was an assertion failure or any other kind of failure. I'll talk with Alfonso and Jordan whether we want to expose assertion errors as you're suggesting. Thanks for pointing this out!

Reply:
I don't think we should change this behavior. The distsql engine also used to swallow errors when a plan was unsupported and fall back to the local execution engine. I'm mostly concerned about users seeing an assertion failure and thinking there's something wrong. `SET vectorize=experimental_always` already scares CockroachDB developers when we fail to read the database name in the prompt.

Reply:
Well, there is something wrong if there is an assertion failure.. If we swallow it, we will never know about it - we'll just have cases where we think vectorized should be used and yet it isn't and we have no idea why. At the very least, these should not be swallowed in tests.